### ISSUE TITLE
"signature not found" - but payment goes through anyway 

### ISSUE THREAD

**Issue by showered** [kinecosystem/kin-ios#16]
We're getting an intermittent issue on iOS whereby we are receiving a client error `("grpc_message":"unhandled error from SubmitTransaction: signature not found","grpc_status":13)`, but the transaction actually does go through anyway.

Further to this (and really important to us) is that the Event Webhook does not get called in this case.

tx: [28DYsuJ3zyds4MUVAnguUneKKmjZgRAr9pFS5S2QrWoCYPo1S7Xh1L353AHdbryxADhD7aebXN64GTzYm8u8A5e2](https://explorer.solana.com/tx/28DYsuJ3zyds4MUVAnguUneKKmjZgRAr9pFS5S2QrWoCYPo1S7Xh1L353AHdbryxADhD7aebXN64GTzYm8u8A5e2)
:white_check_mark: Client success
:white_check_mark: Payment goes through on blockchain.
:white_check_mark: Agora event webhook is fired

tx: [4KQYYBTCGyoSspwrJess7KA9duMf3d8ZVXFuNFhSugjGSBwHu7kESRfwt4M5esq3qV5dm2iCMikWRf7uHnhGCeXe](https://explorer.solana.com/tx/4KQYYBTCGyoSspwrJess7KA9duMf3d8ZVXFuNFhSugjGSBwHu7kESRfwt4M5esq3qV5dm2iCMikWRf7uHnhGCeXe)
:x:  Client error (signature not found)
:white_check_mark:  Payment goes through on blockchain
:x:  Agora event webhook is NOT fired

Full iOS SDK error:

```
transientFailure(error: Error Domain=io.grpc Code=13 "{"created":"@1619040502.719064000","description":"Error received from peer ipv4:54.237.169.129:443","file":"ios/Pods/gRPC-Core/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"unhandled error from SubmitTransaction: signature not found","grpc_status":13}" UserInfo={io.grpc.HeadersKey={
}, io.grpc.TrailersKey={
}, NSDebugDescription={"created":"@1619040502.719064000","description":"Error received from peer ipv4:54.237.169.129:443","file":"ios/Pods/gRPC-Core/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"unhandled error from SubmitTransaction: signature not found","grpc_status":13}, NSLocalizedDescription=unhandled error from SubmitTransaction: signature not found})
```

**Comment by hitwill** [comment]
Thanks @showered this is an issue that is being tracked - the problem arises when Agora doesn't receive a definite response from the blockchain.

A patch is in the works where Agora will return the tx id when the response is ambiguous (not a success or fail) - that will enable the client to confirm the status of the transaction

**Comment by showered** [comment]
Thanks @hitwill that's good to hear. Do you think the webhook issue will also be resolved as part of the same fix? Or will we need to tweak our flow so that the tx id needs to be manually looked up?

**Comment by hitwill** [comment]
@showered - The webhook unfortunately can't be guaranteed in this case. It's actually Solana that gives an ambiguous response so Agora is also uncertain. However, once it returns the tx id to the client, the client (or server) would need to use it to confirm the definite response.

**Comment by samdowd** [comment]
https://github.com/kinecosystem/kin-ios/blob/7df69dc329339f2d088be36ab2aee57b5054fdb0/KinBase/KinBase/Src/KinAccountContext.swift#L532-L536

Wondering if these lines have our hint. At the very least it's a redundant else if, could be more. But it does look like Android has some errors defined that iOS does not.

**Comment by hitwill** [comment]
That's definitely duped code. I wonder if the dev meant to put a different condition there.

**Comment by showered** [comment]
This issue is now resolved in version 2.0.0.

Thank you to everyone involved in providing the fix!
